src/main/kotlin/com/terraformation/backend/tracking/mapbox/MapboxService.kt

```diff
@@ -87,4 +87,6 @@ class MapboxService(
   data class RetrieveTokenResponsePayload(val code: String, val token: Token) {
     data class Token(val user: String)
   }
+
+  data class GeneratedTemporaryToken(val token: String, val tokenExpirationMs: Long)
```

Comment:
The second field here won't be what you're looking for, I think, though I can't say for sure without seeing the client side. This is an `Instant`, not a `Duration`, that is, it is the number of milliseconds since 1/1/1970 rather than the number of future milliseconds until expiration.

Reply:
Was trying to capture the time at which the token expires.
For a token generated at  Mon Nov 14 9:32am, client saw expiration ms as 1668448953057, which is  Mon Nov 14 2022 10:02:33 GMT-0800 (Pacific Standard Time)
I closed this ticket, like you had thought - client can capture mapbox 401s to regenerate the token.

Reply:
Ah, okay, sorry for the comment noise, then. I wasn't sure if you had intended this to be an offset or an absolute time. But glad it turned out to not be needed anyway.